crates/rolldown/src/hmr/hmr_stage.rs

```diff
@@ -275,7 +275,7 @@ impl<'a> HmrStage<'a> {
           .collect::<Vec<_>>(),
       );
 
-      self.cache.merge(module_loader_output.into());
+      self.cache.merge(module_loader_output.into()).map_err(|e| vec![anyhow::anyhow!(e).into()])?;
```

Comment:
`Hmr` could choose how to handle this: trigger a rebuild, or just raise a Diagnostic.